feature_engine/imputation/na_transformer.py

```diff
@@ -0,0 +1,79 @@
+# Authors: Pradumna Suryawanshi <pradumnasuryawanshi@gmail.com>
+# License: BSD 3 clause
+
+from typing import Optional, List, Union
+
+import pandas as pd
+
+from feature_engine.dataframe_checks import _is_dataframe
+from feature_engine.imputation.base_imputer import BaseImputer
+from feature_engine.parameter_checks import _define_numerical_dict
+from feature_engine.variable_manipulation import (
+    _check_input_parameter_variables,
+    _find_or_check_numerical_variables,
+)
+
+
+
+
+class Na_transformer(BaseImputer):
+
+    def __init__(
+        self,
+        missing_only: bool = True,
+        variables: Union[None, int, str, List[Union[str, int]]] = None
+    ) -> None:
+
+        if not isinstance(missing_only, bool):
+            raise ValueError("missing_only takes values True or False")
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.missing_only = missing_only
+
+
+    
+    def return_na(self):
```

Comment:
maybe call this method `return_dropped_data`

This method should be identical to transform(), except that instead of returning the data without NA, it would return the data with NA. So in essence it does the reciprocal of transform. 

So something like this:

```
def return_dropped_data(self, X: pd.DataFrame) -> pd.DataFrame:

        X = self._check_transform_input_and_state(X)

        id=pd.isnull(X[self.variables_]).any(1)
        id=id[id==True]
        return X[X.index.isin(id.index)]
```

Reply:
but actually, maybe instead of isin, use pandas loc, or iloc.

Reply:
is there not a way to retain observations with na with pandas, without the need to create an intermediate index?